### DISCUSSION TITLE
Is it possible to configure Herd/nginx to accept connection on IP:PORT instead of URL?

### DISCUSSION THREAD

**Discussion by jtomlinson** [beyondcode/herd-community#626] [Q&A]
I need to be able to connect a local android device to my development environment. Previously, on Linux, using apache/haproxy, I was able to configure a connection using an IP and port (http://192.168.1.46:8080) to allow the external device to connect to my laravel dev environment. I know I can use Expose to make this connection, but since I don't actually need to serve the site outside of my local network it would be nice to not have to use Expose if possible.

**Comment by mpociot** [comment]
The easiest way, without breaking any additional sites, would be to run `herd isolate [PHP-VERSION]` for this specific site.
This generates a custom nginx configuration file that will be loaded when nginx starts.

There you can pretty much customize everything you want. The file itself can be found here: `~/Library/Application Support/Herd/config/valet/Nginx/`

**Reply by jtomlinson** [reply]
This got me partway there. I am able to access http://ipaddress, but seeing the Herd 404 page. Do I need to configure something in Herd itself to serve my site from the IP address?

![Screenshot 2024-04-10 at 9 29 58 AM](https://github.com/beyondcode/herd-community/assets/372896/c9732dbf-b8b1-4348-94ee-b8cb328639d2)

**Comment by jtomlinson** [comment]
I was able to get this working by adding my local IP to `server_name` in the nginx configuration file.

`server_name site.test 172.16.1.58;`

And then in my .env file I changed `APP_URL` from `site.test` to `172.16.1.58` and was able to access the site, by IP, locally and from an android device.
